application/apps/indexer/addons/file-tools/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Idea is to get the file path from the user,
+// read the first chunks from the file and return String
+// 'binary' or 'text' as of now. Later we can add enum for file
+// type.
+
+use anyhow::Result;
+use std::{fs::File, io::Read, path::Path};
+
+const BYTES_TO_READ: usize = 10240;
+
+pub fn detect_file_type(file_path: &Path) -> String {
```

Comment:
Actually you are not detecting type of file... It's a little confuse, because by type we understand as usual `executable, image, video etc.`... I would suggest to have as public methods `is_text()` and `is_binary()`